zuul-core/src/main/java/com/netflix/zuul/message/Headers.java

```diff
@@ -548,19 +624,50 @@ private void truncate(int i) {
         }
     }
 
+    /**
+     * Checks if the given value is compliant with our RFC 7230 based check
+     */
+    private boolean isValid(String value) {
+        if (value == null || findInvalid(value) == ABSENT) {
```

Comment:
I dont think null values should ever reach here.  All public methods check their params before using them, so it shouldn't be necessary to double check here.

Reply:
Some public methods (like set API and its variants) allow for null as header values. This was the reason for having the null check for isValid and validateField.

Reply:
Then it needs to be specified on this method as `@Nullable`

Reply:
Makes sense. Changed it.